metal/devices.go

```diff
@@ -96,11 +103,25 @@ func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloud
 	}, nil
 }
 
-func nodeAddresses(device *packngo.Device) ([]v1.NodeAddress, error) {
-	var addresses []v1.NodeAddress
-	addresses = append(addresses, v1.NodeAddress{Type: v1.NodeHostName, Address: device.Hostname})
-
-	var privateIP, publicIP string
+func nodeAddresses(device *packngo.Device, providedNodeIP string) ([]v1.NodeAddress, error) {
+	var (
+		addresses           []v1.NodeAddress
+		unique              = map[v1.NodeAddress]bool{}
+		privateIP, publicIP string
+	)
+	addr := v1.NodeAddress{Type: v1.NodeHostName, Address: device.Hostname}
+	unique[addr] = true
+	addresses = append(addresses, addr)
+
+	// if the kubelet was started with --node-ip, that must be the first address to use
+	if providedNodeIP != "" {
+		privateIP = providedNodeIP
+		addr = v1.NodeAddress{Type: v1.NodeInternalIP, Address: providedNodeIP}
+		if _, ok := unique[addr]; !ok {
+			unique[addr] = true
+			addresses = append(addresses, addr)
+		}
+	}
 	for _, address := range device.Network {
 		if address.AddressFamily == int(metadata.IPv4) {
```

Comment:
```suggestion
		if providedNodeIP == address.Address {
			continue
		}
		if address.AddressFamily == int(metadata.IPv4) {
```

The `unique` map would have the public `providedNodeIP` address indexed as `NodeInternalIP` and if we don't do something like the above we'll add a second index for the same address as `NodeExternalIP`.

(GKE permits public internalips too https://cloud.google.com/kubernetes-engine/docs/how-to/alias-ips#enable_pupis)

Reply:
Maybe it makes more sense to have the `unique` map index on the address string, rather than the `v1.Address` struct? That way it guarantees uniqueness?

```go
// partial code
var 		unique              = map[v1.NodeAddress]bool{}
	addr := v1.NodeAddress{Type: v1.NodeHostName, Address: device.Hostname}
	unique[addr.Address] = true
// etc.
```

Reply:
Done.

Reply:
I don't see this change committed.

I would expect to find:
```go
var unique = map[string]bool{}
addr := v1.NodeAddress{Type: v1.NodeHostName, Address: device.Hostname}
unique[addr.Address] = true
```

But instead I see: https://github.com/equinix/cloud-provider-equinix-metal/pull/297/files#diff-3ddaf4333cb941b353834040ab733477c56a207db5bcf2e854b62fe663db5494R109-R113

(maybe I'm github'ing incorrectly)

Reply:
Aha! I had a failed push due to bad network, which I hadn't realized. Pushed now and updates.

Reply:
> githubing

Is that a verb now? 😁